src/imlac/maze.print

```diff
@@ -0,0 +1,81 @@
+; " -*- muddle -*- "
+
+; "Print an IMTRAN'ed maze.'
+
+<SETG MAZE <IVECTOR 34>>
+<SETG WEST [!<IVECTOR 13 " "> "W" "E" "S" "T" !<IVECTOR 15 " ">]>
+<SETG EAST [!<IVECTOR 13 ""> "     E" "     A" "     S" "     T"
+	    !<IVECTOR 15 "">]>
+
+<DEFINE GET4 (CH "AUX" (C <ASCII <READCHR .CH>>))
+  ;"Read 4 bits from an imtraned file."
+  ;"Ignore all characters outside @-O."
+  <COND (<AND <G=? .C 64> <L=? .C 79>> <- .C 64>)
+	(ELSE <GET4 .CH>)>>
+
+<DEFINE GET8 (CH)
+  <+ <* <GET4 .CH> 16> <GET4 .CH>>>
+
+<DEFINE GET16 (CH)
+  <+ <* <GET8 .CH> 256> <GET8 .CH>>>
+
+<DEFINE GET-BLKLDR (CH)
+  ;"Skip over the block loader."
+  <REPEAT ((N 0))
+    <GET16 .CH>
+    <SET N <+ .N 1>>
+    <COND (<=? .N 65> <RETURN>)>>>
+
+<DEFINE READ-MAZE (CH)
+  ;"Read an imtraned maze."
+  <GET-BLKLDR .CH>
+  <REPEAT ((I 1) N)
+    ;"Get block length."
+    <SET N <+ <GET8 .CH> .I>>
+    ;"Get block address.  All ones means end."
+    <COND (<=? <GET16 .CH> 65535> <RETURN>)>
+    <REPEAT ()
+      <PUT ,MAZE .I <GET16 .CH>>
+      <SET I <+ .I 1>>
+      <COND (<=? .I .N> <RETURN>)>>
+    ;"Get block checksum."
+    <GET16 .CH>>>
+
+<DEFINE WRITE-LINE (CH S)
+  <PRINC .S .CH>
+  <TERPRI .CH>>
```

Comment:
You could make CH implicit, then you don't need to put in in each subordinate routine call. 
(cf. "The Mudde Programming Language" Section 24.1.1 Comment 4.)
<DEFINE WRITE-LINE (S OUTCHAN)
    #DECL ((S) ANY (OUTCHAN) <SPECIAL CHANNEL>)
    <PRINC .S>
    <TERPRI>>